### DISCUSSION TITLE
Custom Trace Id Generation

### DISCUSSION THREAD

**Discussion by perhapsmaple** [open-telemetry/opentelemetry-cpp#3192] [Q&A]
Hello,

I'm working on a distributed system where my application is split into several disconnected processes. Because these processes are not directly connected, I have no way to propagate my trace context and have to dynamically generate a trace id based on the file being processed. I am considering using MD5 to generating my trace id by hashing the name of the file being processed. 

Is there an existing way that I can plug in this functionality into the SDK so that I can allow my trace id to be dynamically generated based on external parameters instead of using the default RandomGenerator? Is this the right approach for tackling this situation with completely disconnected processes?

Any help or improvements will be appreciated
Thanks

**Comment by lalitb** [comment]
If you are looking to plugin your own random generator for traceid and spanid, it can be added through TracerProvider constructor - https://github.com/open-telemetry/opentelemetry-cpp/blob/23562e6da33155419c84e3f62254c738d74daa09/sdk/include/opentelemetry/sdk/trace/tracer_provider.h#L48

As an example, ETW exporter uses it's custom random-id generator - https://github.com/open-telemetry/opentelemetry-cpp/blob/23562e6da33155419c84e3f62254c738d74daa09/exporters/etw/include/opentelemetry/exporters/etw/etw_random_id_generator.h#L18

Note - please note the MD5 hash mayn't be ideal random id generator - the recommendation is to have one with true uniqueness and high entropy, with uniformly distributed over the entire 128-bit space. This is to ensure that sampling algorithms work properly over them.
